planner/core/initialize.go

```diff
@@ -404,6 +406,15 @@ func (p PhysicalIndexJoin) Init(ctx sessionctx.Context, stats *property.StatsInf
 	return &p
 }
 
+// Init initializes PhysicalIndexMergeJoin.
+func (p PhysicalIndexMergeJoin) Init(ctx sessionctx.Context) *PhysicalIndexMergeJoin {
+	ctx.GetSessionVars().PlanID++
```

Comment:
We can call `p.PhysicalIndexJoin.Init(ctx)`, then reset `p.tp`.

Reply:
`PhysicalIndexJoin.Init(ctx)` `new`s a new `basePlan`. I think that's unnecessary.